app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/SubscriptionController.scala

```diff
@@ -63,33 +64,45 @@ class SubscriptionController @Inject()(
       val result = for {
         subscriptionResponse <- subscriptionService.subscribe(details)
         _ <- EitherT(
-              updateSession(sessionStore, request)(
-                _.copy(
-                  journeyStatus = Some(
-                    Subscribed(
-                      SubscribedDetails(
-                        details.name,
-                        details.emailAddress,
-                        details.address,
-                        details.contactName,
-                        CgtReference(subscriptionResponse.cgtReferenceNumber),
-                        None,
-                        registeredWithId = true
+              subscriptionResponse match {
+                case SubscriptionResponse.SubscriptionSuccessful(cgtReferenceNumber) =>
+                  updateSession(sessionStore, request)(
+                    _.copy(
+                      journeyStatus = Some(
+                        Subscribed(
+                          SubscribedDetails(
+                            details.name,
+                            details.emailAddress,
+                            details.address,
+                            details.contactName,
+                            CgtReference(cgtReferenceNumber),
+                            None,
+                            registeredWithId = true
+                          )
+                        )
                       )
                     )
                   )
-                )
-              )
+                case SubscriptionResponse.AlreadySubscribed =>
+                  updateSession(sessionStore, request)(
+                    _.copy(journeyStatus = Some(AlreadySubscribedWithDifferentGGAccount))
+                  )
+              }
             )
       } yield subscriptionResponse
 
       result.fold(
         { e =>
           logger.warn("Could not subscribe", e)
           errorHandler.errorResult()
-        }, { subscriptionResponse =>
-          logger.info(s"Successfully subscribed with cgt id ${subscriptionResponse.cgtReferenceNumber}")
-          SeeOther(routes.SubscriptionController.subscribed().url)
+        }, {
+          case SubscriptionResponse.SubscriptionSuccessful(cgtReferenceNumber) =>
```

Comment:
Worth importing the values of `SubscriptionResponse` unless you think the semantics are clearer this way

Reply:
I had no real reason of diong it this way - I've changed it now